r/tests/testthat/test-dplyr-lubridate.R

```diff
@@ -32,97 +32,132 @@ if (tolower(Sys.info()[["sysname"]]) == "windows") {
 } else {
   test_date <- as.POSIXct("2017-01-01 00:00:12.3456789", tz = "Pacific/Marquesas")
 }
-test_df <- tibble::tibble(date = test_date)
-test_date_df <- tibble::tibble(date = as.Date(as.character(test_date)))
 
-test_that("extract year from date32 objects", {
-  expect_equivalent(
-    test_date_df %>%
-      mutate(x = year(date)) %>%
-      collect() %>%
-      select(x),
-    test_df %>%
-      mutate(x = year(date)) %>%
-      collect() %>%
-      select(x)
+test_df <- tibble::tibble(
+  datetime = test_date,
+  date = as.Date("2021-09-09")
+)
+
+test_that("extract year from timestamp", {
+  expect_dplyr_equal(
+    input %>%
+      mutate(x = year(datetime)) %>%
```

Comment:
It looks like none of these tests are testing the `date32` path. Shouldn't they all look something like this?

```suggestion
      mutate(
        year(datetime),
        year(date)
      ) %>%
```

Reply:
So the earlier tests in the file are testing datetimes aka timestamps whereas the later ones are testing dates/date32 objects.  I've added in a couple of comments to make this more clear.